src/client/Microsoft.Identity.Client/Internal/Requests/InteractiveRequestOrchestrator.cs

```diff
@@ -0,0 +1,143 @@
+            }
+
+            PublicAuthCodeRequest webUiInteractiveRequest = new PublicAuthCodeRequest(
+                _requestParams,
+                _interactiveParameters,
+                authCode,
+                pkceCodeVerifier);
+
+            return await webUiInteractiveRequest.RunAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<AuthenticationResult> FetchTokensFromBrokerAsync(string brokerInstallUrl, CancellationToken cancellationToken)
+        {
+            IBroker broker = _serviceBundle.PlatformProxy.CreateBroker(
+                _interactiveParameters.UiParent);
+
+            BrokerInteractiveRequest brokerInteractiveRequest = new BrokerInteractiveRequest(
+                _requestParams,
+                _interactiveParameters,
+                broker,
+                brokerInstallUrl);
+
+            return await brokerInteractiveRequest.RunAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        private IWebUI CreateWebAuthenticationDialog()
+        {
+            if (_interactiveParameters.CustomWebUi != null)
+            {
+                return new CustomWebUiHandler(_interactiveParameters.CustomWebUi);
+            }
+
+            CoreUIParent coreUiParent = _interactiveParameters.UiParent;
+
+            coreUiParent.UseEmbeddedWebview = GetUseEmbeddedWebview(
+                _interactiveParameters.UseEmbeddedWebView,
+                _serviceBundle.PlatformProxy.UseEmbeddedWebViewDefault);
+
+#if WINDOWS_APP || DESKTOP
+            // hidden web view can be used in both WinRT and desktop applications.
+            coreUiParent.UseHiddenBrowser = _interactiveParameters.Prompt.Equals(Prompt.Never);
+#if WINDOWS_APP
+            coreUiParent.UseCorporateNetwork = _serviceBundle.Config.UseCorporateNetwork;
+#endif
+#endif            
+            return _serviceBundle.PlatformProxy.GetWebUiFactory()
+                .CreateAuthenticationDialog(coreUiParent, _requestParams.RequestContext);
+        }
+
+        private static bool GetUseEmbeddedWebview(WebViewPreference userPreference, bool defaultValue)
+        {
+            switch (userPreference)
+            {
+                case WebViewPreference.NotSpecified:
+                    return defaultValue;
+                case WebViewPreference.Embedded:
+                    return true;
+                case WebViewPreference.System:
+                    return false;
+                default:
+                    throw new NotImplementedException("Unknown option");
+            }
+        }
+
+        private bool IsBrokerRequiredAuthCode(string authCode)
+        {
+            if (authCode.StartsWith(BrokerParameter.AuthCodePrefixForEmbeddedWebviewBrokerInstallRequired, StringComparison.OrdinalIgnoreCase)) 
+               //|| authCode.StartsWith(_serviceBundle.Config.RedirectUri, StringComparison.OrdinalIgnoreCase) // TODO: what is this?!
```

Comment:
@jennyf19 - I don't understand this line - can this ever be true?

Reply:
yes, if the user has a wpj requirement, the auth code will come back w/msauth:// at the beginning.

---
In reply to: [374247211](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1611#discussion_r374247211) [](ancestors = 374247211)